### ISSUE TITLE
Auto Reload breaks after the first target is killed

### ISSUE THREAD

**Issue by smcclennon** [IceHacks/KrunkerCheatInjector#23]
Weapons used: Sniper, Semi-Auto & default pistol

![Settings](https://i.imgur.com/pqAmGww.png)

Once the first target is found (and is constantly visible), weapons will fire at them until the mag is empty, and then reload and continue firing (assuming the target is still alive)

If the original target is no longer visible (goes behind a building, is killed by yourself or another play, leaves etc), your weapon will reload after each shot (only one bullet will leave the mag before a reload).

TLDR; If the original target dies or is no longer visible, Auto Reload reloads your weapon after each individual bullet leaves the mag.

**Comment by IceHacks** [comment]
This is expected. If there are no enemies in its view and you have less than max ammo it will reload. It doesn't reload when ammo reaches `0` it reloads when ammo less than max.
